### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by vr8hub** [barnesstorming/andre-norton_storm-over-warlock#1]
- [x] As noted on the list, there are a number of unsemanticated italics (`<i>`) that need to be corrected, either to emphasis (`<em>`), or to add semantics. See step 12 of the Step-by-Step and section 8.2 of the manual.
- [x] The "slap-slap" in chapter 14 is a sound, so it actually should be unsemanticated italics. (See SEMOS 8.2.6.)
- [x] As also noted on the list, modernize spelling changes, which are editorial, were mixed into a non-editorial commit. They will need to be rebased out into their own commit, marked as `[Editorial]`.
- [x] Modernize-spelling also makes other changes now that should be accepted.
- [x] All of the CSS from section 7.5 of the manual needs to be added. Lint will tell you if there is any that is unused.
- [x] Since the text says the "verse" in chapter 13 is a chant, they should probably be z3998:song instead of verse. (The CSS should then be adjusted as well.)
- [x] I'm guessing LN in the first transcriber's name are initials, since both are capitalized. If so, they should be formatted as such (L. N.) and tagged separately. Looking in the corpus, the outer `<b epub:type="z3998:personal-name">` stays, just an additional `<abbr epub:type="z3998:given-name">` would be wrapped around the initials (collectively).
- [x] Alice's full (actual) name should be in the metadata; see [SEMOS 9.11.4](https://standardebooks.org/manual/1.6.4/single-page#9.11.4) for the format. I see that is missing from our other Norton books; we'll get those fixed as well.
- [x] The cover.source.jpg in /images should be the original image, not the uprezzed one (see [SEMOS 2.1.3.3](https://standardebooks.org/manual/1.6.4/single-page#2.1.3.3)). That way if the ability to improve the uprez arises in the future, we have the original source image to operate on.
- [x] The long description mentions that this is part of a series. If so, then there should be series information in the metadata; see [SEMOS 9.9.3.3](https://standardebooks.org/manual/1.6.4/single-page#9.9.3.3). (You can also look at other series' books in the corpus, e.g. Tarzan, Sherlock Holmes, Lecoq, Scarlet Pimpernel, etc.)

**Comment by barnesstorming** [comment]
It's a longer punchlist than I'd hoped for, but I'll get on it, thanks. For the third bullet, the modernize spelling issue, where you say changes "will need to be rebased out into their own commit, marked as [Editorial]" -- functionally how do I do that, since that's so many iterations of the book back?

**Comment by vr8hub** [comment]
You can do a web search on "rebase interactive" and get the details; there have also been a few discussions on the list, so you can search there as well. Essentially, you choose which commit you want to rebase, and git puts the repository back to that point in time, and then you make whatever changes need to be made, commit them, and then tell it to roll everything else forward.
You'll want to make a copy (or two) of your local repository before you try anything (it usually takes a try or three the first time), and don't push the changes to GitHub until you're sure they're OK.

**Comment by vr8hub** [comment]
[Here's](https://groups.google.com/g/standardebooks/c/Q96zngDIwds/m/IT05HsUuDAAJ) something I posted to someone else on the list about this.

**Comment by barnesstorming** [comment]
I think I've got the rebase bit nailed successfully. Fingers crossed. Thx for the guidance.

Re the item about CSS from section 7.5, I'm not getting something. I've added all the code, as you suggest. And as you predicted, LINT is telling me everything I added is unused. Do I leave it there and ignore the LINT nags or delete the unused CSS back out? (I'm not understanding its presence if it's unused...)

**Comment by barnesstorming** [comment]
Do you think it's safe to just assume that LN are the transcribers initials for the purpose of formatting? It seems unlikely that someone's name is "LN" but theoretically possible. (The book came thru the Online Distributed Proofreading Team, and I've searched their forums at https://www.pgdp.net for Mr/Mrs Yaddanapudi without success.)

**Comment by vr8hub** [comment]
Re the poetry/verse/song CSS, _all_ of it should not be errorring. You don't need the i1/i2 classes, and the `p + p` since there aren't multiple stanzas, but everything else is needed. If it's errorring there's a different problem.

Re the initials, while it _could_ be a name, in that case I would expect to see it as Ln, not LN. Since it's two caps, I think assuming initials is a reasonable assumption.

**Comment by vr8hub** [comment]
And while I was asking about updating the other Norton's for her name, I learned that we're not supposed to use real names in the full name metadata (which makes the Twain I looked at incorrect), so you should use her full pseudonym, i.e. Andre Alice Norton. Sorry for the confusion, I still learn something every day! :)

**Comment by vr8hub** [comment]
Lint is complaining about the CSS because you changed the type to "song" in the text, but didn't change it in the CSS.

**Comment by barnesstorming** [comment]
The version on Github doesn't reflect the most recent changes -- which I've held open pending the earlier questions. I can push that if you want to see what I'm talking about, but the error log LINT is giving me is "Unused CSS selectors" for:

[epub|type~="z3998:poem"] p    
[epub|type~="z3998:poem"] p + p 
[epub|type~="z3998:poem"] p > span
[epub|type~="z3998:poem"] p > span + br      
p span.i1     
p span.i2

**Comment by vr8hub** [comment]
Yes, see my comment right before yours. :)

**Comment by barnesstorming** [comment]
There is no song/verse discrepancy in my version. The current CSS is: 

@charset "utf-8";
@namespace epub "http://www.idpf.org/2007/ops";

[epub|type~="z3998:poem"] p{
	text-align: initial;
	text-indent: 0;
}

[epub|type~="z3998:poem"] p + p{
	margin-top: 1em;
}

[epub|type~="z3998:poem"] p > span{
	display: block;
	padding-left: 1em;
	text-indent: -1em;
}

[epub|type~="z3998:poem"] p > span + br{
	display: none;
}

p span.i1{
	padding-left: 2em;
	text-indent: -1em;
}

p span.i2{
	padding-left: 3em;
	text-indent: -1em;
}


[epub|type~="z3998:song"] p{
	text-align: initial;
	text-indent: 0;
}

**Comment by barnesstorming** [comment]
Should every iteration of "poem" be "song" instead? I'm sorry to be obtuse about this. I just don't understand.

**Comment by vr8hub** [comment]
There is—your _text_ (in chapter 13) is using an epub:type of "z3998:song", so your CSS has to target "z3998:song" as well. The CSS is saying "when I find anything with an epub:type of z3998:poem, then format it like this." But you don't have anything with an epub:type of z3998:poem. You have things with an epub:type of z3998:song, so your CSS has to target that epub:type. If you had both verse _and_ song, then your CSS would have to target both. And so forth.

**Comment by barnesstorming** [comment]
I get that. But do I simply delete the references to CSS "poem" formatting, leaving only the single reference to "song", or should each be changed to "song" with the CSS commands left in place? ... This is just a lack of understanding of HTML for me, I believe. (Thank you for your patience. I swear I'm not an idiot about most things.)

**Comment by vr8hub** [comment]
One of the many things I had to learn when I started with SE was how CSS worked. I'm still a neophyte compared to Alex and Robin, but I can get around a lot better than when I started. Like git, there are dozens of places to learn CSS on the web. Mozilla's documentation on [CSS selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) is good once you have the basics down.

**Comment by vr8hub** [comment]
No, the single reference to "song" is what started this discussion. :) All of those CSS selectors are targeting a specific epub:type (z3998:poem), and they need to be targeting what you actually have, which is z3998:song. So all of the "poem"s need to be changed to "song". (Which then means you can get rid of the existing "song" one, as it is now a duplicate.) As I mentioned earlier, you don't need the i1/i2 targets; lint will still complain about those even after the above changes.

**Comment by barnesstorming** [comment]
Gotcha. So each piece of that CSS is formatting a *portion" of a poem/song/whatever, or different cases of them, if I'm sussing it out correctly. (I'll put "tackle understanding CSS" on my to-do list down the road...) Making that change and moving on. Thx again.

**Comment by barnesstorming** [comment]
OK, that should be everything on that initial punchlist, with one question: 

When I added the collection information, I used SE's other Norton collections as a guide. They both use the id "collection-1" despite the fact that they're in different series. ... Should the collection number (in my metadata, that's also currently id="collection-1") be distinct from the other collections, or does that not matter between different ebooks so long as all the books in a single collection share the same "friendly" name after the bracket?

**Comment by vr8hub** [comment]
The "collection-1" is just the id on the metadata, just like "transcriber-1", "subject-1", etc. It is specific to this set of metadata, i.e. this book. It's the "belongs-to-collection" property that defines the series name, and the "group-position" that defines this book's place within the series.

**Comment by vr8hub** [comment]
The beefa92 commit ("Add titles") still has a number of spelling changes in it, e.g. eyewitness, newfound, etc. It looks like you did two things in that one commit: run modernize-spelling, and run build-titles. If that's all you did, i.e. no manual changes, then it shouldn't be too difficult to separate them. (As before, and _always_, make a backup or two of your repository before trying this.)
You can run a rebase interactive, editing the b33fa92 commit, which will drop you in _after_ the add titles changes.
You then want to undo the most recent change, i.e. the adding of titles, getting rid of the changes. In git, this is called a hard reset. You can web search for "git hard reset"; I've used git-extras so long I don't remember how do it the hard way. :)
You'll now be sitting after the third commit (Semanticate). Now you can do the two things (modernize spelling, build titles), committing each one _separately_. Then you can continue the rebase to finish it. The end product looks the same, but now you have two commits where you had one before, with modernize spelling properly noted as Editorial.

**Comment by barnesstorming** [comment]
Well, I've spent about 90 minute doing this and fear I might have done something wrong. The added commits aren't showing up; there's just an added "Merge branch ..." notice at the top, with no details on all the commits I added. Ugh. I feel like I'm officially wasting both of our time, Vince. Any advice on a best next step?

**Comment by vr8hub** [comment]
Definitely not wasting time, just part of the learning process. :) Don't worry about it.

Sounds like you're using something other than the command-line for git?

**Comment by barnesstorming** [comment]
I'm using the command line. It was behaving a bit strangely -- like, acting as if the SE commands weren't making changes in some cases, and "git diff" wasn't showing anything, even though they *had* made changes in previous iterations. But then some of the tools *were* making changes, so I wrote it off as "maybe their were no archaic spellings" and whatnot. ... When I tried to push the new commits thru, Chapter 14 wouldn't merge and things sort of went off the rails. Just to get the repository back up to date and resumed, I had to force that file through. Suffice it to say I went well outside of my comfort zone with both github and command-line Linux, both of which are new to me. (There was "push"ing and "pull"ing and even "cherry-picking" involved, and other scary stuff (to me).)

According to my log, in that earlier commit that's causing all the trouble commit, I did this:
* ran modernize-spelling (reviewing individual changes)
* ran find-mismatched-diacritics (finding none)
* ran find-mismatched-dashes and and made several changes
* set title elements.

I'm thinking maybe I'll just roll it back one in github and then try rebasing again? 

Failing that, I can roll all the way back to that second commit and start it over from there, I suppose.

**Comment by vr8hub** [comment]
Where did you see the "Merge branch" notice? That's why I asked about the command-line.
Yes, it's all scary stuff until we learn a bit more about it. :) It took me quite a bit of time with git before I tried to tackle one of these.

If you're doing this locally (and that's where you want to be doing it), then GitHub need not (should not) be involved. So I'm not sure what you mean by "roll back one in GitHub". You made a backup of the repository before starting like I told you, right? Then you can just copy the backup and start again on the copy.

We only want to do here what was done originally. Were the find-mismatched-dashes run in this commit before? I didn't notice them if so, but I wasn't looking specifically for them. If not, we don't want to do them now. If you did, we do.

The steps are:
1. `git rebase --interactive` (with the hash of the commit before the one we're editing). For the local clone I made from your repository, that is the Semanticate commit, i.e. bb1f92b.
2. In the editor, change the "pick" on the first commit (which should be the "Add titles" commit, i.e. beefa92; if it isn't, then the wrong one was chosen in step 1 to "edit".
3. Save and exit the editor.
4. It will drop you into the repository in the state it was in _after_ the "Add titles" commit. But we want to get rid of that commit entirely.
5. Do the git reset. Looking at git undo source, it looks like this is `git reset --hard HEAD~1`. (Do a hard reset on the most recent commit on HEAD). Now the repository is in the state it was after the semanticate commit. Neither the titles nor spelling has been changed.
6. `se modernize-spelling` Commit it (with [Editorial]).
7. `se build-title` You'll get an error; chapter 15 is missing it's top `<section`, so you'll have to add it. It should look just like all the other chapter sections, but with chapter-15 as the id.
8. `se build-title` again.
9. If you ran `se find-mismatched-dashes` in this commit originally, then run it and commit it. Note that since this is also changing content (not formatting), it should also be an editorial commit.
10. `git rebase --continue` to finish the rebase. It will roll everything forward unless it hits a merge error.
11. Sure enough, you'll get a merge error on chapter 14. This is because there was a space following the `</hgroup>` in that file, and so when you ran "Clean and Lint" in a later commit, it jumbled that file. But now that file has two conflicting changes between the original and the rebase version: the jumbling that the "Clean and Lint" commit did, and the changes to spelling that the modernize-spelling commit made.
12. If you edit chapter-14, you'll see a <<<< marker with what's in HEAD (the formatted, rebase version), a ==== marker separating the two, and then a >>>>> marker indicating the end of the second one. When a merge conflict happens, you have to pick which one you want to keep and make the changes from the other one. In this case, we want the jumbled one (the bottom one) because that's the way it looked originally. But, we need to make the spelling change to that one; chapter 14 only has a single change, semi-liquid to semiliquid. So, you can delete the top one, i.e. what's between the first and second markers, _including the markers_. Then change semi-liquid to semiliquid in the bottom one. Then delete the bottom marker and save the file. When you're done, you should only have the jumbled one, and there should be no markers left.
13. `git add chapter-14.xhtml` (you might need the directory if you're not in the src/epub/text subdirectory). We have to tell git that we made a change to the file.
14. `git rebase --continue` again. It will pull up the commit message for this commit; just exit and save, as you want to keep the existing message.
15. You'll get another merge conflict on chapter 14. This is when the jumbled part was correctly formatted (after you fixed the space following the hgroup, I presume). Exact same steps—delete the top one, change semi-liquid, delete the bottom marker, save the file.
16. `git add chapter-14.xhtml`
17. git rebase --continue
18. That will roll forward without error.

And with that, you're done. Yes, it is very intimidating; believe me, I didn't attempt it until the not so recent past. This is why we say follow the Step-by-Step guide EXACTLY. :)

**Comment by barnesstorming** [comment]
I got in the habit of pushing to github after every commit after my first failed attempt at creating this ebook, where I felt that every commit wasn't being reflected on GH. (This was likely user error, of course.) ... So yeah, I pushed the attempted rebase to GH yesterday, and that's where I'm seeing the merge message.

I backed up the files before I started the rebase but did not back up the full repository -- because I couldn't determine how to do so. There seems to be a lot of confusion even among GH power users on how to do this. But I think just rolling back one commit will get me where I need to be to try the rebase again, since my attempted rebase doesn't appear to have changed anything. ... Rather, the attempted rebase doesn't seem to appear at all. So backtracking one spot should remove the merge, right?

Thx for the detailed steps. I'm hoping to have a chance to do this this afternoon, though it might not be until tomorrow.

**Comment by vr8hub** [comment]
Well, that makes things more complicated. The time to say you weren’t sure how to back it up was BEFORE you started. :)

I wasn’t talking about GitHub. I was talking about your local repository. Backing it up is as simple as copying the folder to somewhere else. Once done, you essentially have two backups: the extra local copy, and GitHub. You can then play with the local copy all day and do all kinds of mayhem without causing any damage. In a worse-case scenario, you delete the folder, copy the copy back to the original directory, and try again. In all cases, you’re just making changes to your local repository, so your remote (GitHub) copy stays pristine. Once you’re satisfied that everything worked, then you push to GitHub. We don’t want to push to GitHub until we’re sure everything worked. (GitHub’s reason for existence is distributed version control; I promise it’s not losing commits. :)

So, I don’t know what state your local or remote copy is in. I have a copy of your original, before you got started, which I made for the review. I’ll copy it off and make a new clone from your current state when I get a chance, but it won’t be until later this afternoon; I have a long appt coming up.

Regardless, no worries: in a catastrophic scenario, I have a copy of your original that I can email you.

**Comment by barnesstorming** [comment]
Wait wait. I made a local duplicate of the working folder before I did it. I just wasn't sure if there was a GitHub method -- and wasn't sure if the local repository would have all the history that the GH site had.

**Comment by vr8hub** [comment]
That’s good, but as I said, no worries, as I definitely have a copy, too. Don’t touch your copy. :)

If you want to try on your current local (NOT the copy) before I get back, go ahead. I’ll see where things are then. It will probably be 4-5 hours from now.

**Comment by barnesstorming** [comment]
Yep, no rush. I won't be able to do anything with it before tonight or tomorrow morning anyway.

**Comment by barnesstorming** [comment]
So, when I view the lastest commit on github, it says, "[Merge branch 'main' of](https://github.com/barnesstorming/andre-norton_storm-over-warlock/commit/c2ecc61e4881420c953c8f8376a2f6f069f1fca9) [https://github.com/barnesstorming/andre-norton…](https://github.com/barnesstorming/andre-norton_storm-over-warlock) 
…_storm-over-warlock into main"

When I look at the details of the commit it shows 0 changed files, 0 additions, 0 deletions.

When I dial down into the files, though, it *is* showing changes and additions, with notations (and, notably, my attempts to push a later version of Ch. 14 through to get the "debase resume" to complete). 

I'm a little worried about the status of these files now and think I'd be more comfortable with starting with the backup files and trying your rebase steps again. ... If you agree, what's the proper github protocol for replacing the current files with the ones that both of us backed up before I attempted rebasing?

**Comment by vr8hub** [comment]
Sorry, today was a blur of activity.

Although I use GitHub, it’s mostly just as the last step until I hand a book over to be reviewed. So I’m far from knowledgable about its intricacies. I would try:
1. Copying your backup over to your original folder, then doing a force push from there, i.e. git push -f origin main, assuming the alias for your GitHub remote is origin, and your main branch is called main. If either of those aren’t true, substitute the appropriate names. If that works, you’re done, and can begin anew with step 3.
2. If that doesn’t work, you can rename the repository on GitHub, create a new repository with the original name, run the command that GitHub shows you to associate your local repository with the new one (same one you ran originally), and then do a normal push. It should update the new repository with all of your existing local commits.
3. At that point (whether you need just step 1 or both steps), your local and remote should be in sync, and you can start again with the interactive rebase (after ensuring that you still have at least one backup copy of your local repository). I would do all of the rebase steps locally, making sure everything worked and it's in a good state when you’re finished, before pushing again to GitHub. You’ll have to do a force push since you did a rebase.

(Edit: the step numbers didn't come through on email.)

**Comment by barnesstorming** [comment]
No worries. I'm crazy right now too. I should have a window to tackle this Saturday or certainly Sunday.

**Comment by barnesstorming** [comment]
OK, so I've wrangled with this for a few hours and think (thought?) it was all the way thru. Had to repeat steps 10-13 a second time to make the 'slap-slap' change. After github told me "Successfully rebased and updated refs/heads/main" I thought I was home free. But when I tried to push everything to the github online repository, I got: 

 ! [rejected]        main -> main (non-fast-forward)
error: failed to push some refs to 'https://github.com/barnesstorming/andre-norton_storm-over-warlock.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

Any idea what's up? I'm tempted just to do a "git pull" here, but am a little scared about corrupting all those changes I've made.

**Comment by vr8hub** [comment]
Remember, I said you have to do a force push because you did a rebase, i.e. include `-f` in the push. `git push -f origin main`

**Comment by barnesstorming** [comment]
Ah yes. ... And the choirs sing. That seems to have gotten it. And I'm seeing the two separate commits for modernize spelling and build titles in the history now.

**Comment by vr8hub** [comment]
Excellent! Just a few things left:
- [x] The fix-mismatched-diacritics commit is changing content, so it should also be [Editorial]. This will require another rebase interactive, but it's _much_ easier, as you're only changing the commit message. :) Make another copy of your local repository (always!), then `rebase --interactive 33f0a30` which is the commit prior to the one you want to change. It brings up the list; this time, instead of changing the "pick" to "edit", you want to change it to "reword" (you can read below what the various words mean). Then save the file; it will then open the editor again with the commit message for that commit, just put [Editorial] at the beginning of the line and save. It will then rewind everything and that's it. As before, you'll need to do a force push to get it to GitHub. 
- [x] You have a stray word-joiner (\x2060) at the end of the `</hgroup>` line in chapter-14. It's invisible unless you have invisibles turned on in whatever editor you're using. It's immediate following the closing bracket.
- [x] There's also a spare space at the beginning of the line of the first paragraph in chapter-14. ("Something ahead!"). It's the first character on the line, i.e. before the tabs and the `<p>`.
- [x] Once you've fixed those, you'll need to run `se clean` again. It makes other changes that should be accepted.
- [ ] And then modernize-spelling makes other changes as well. Don't forget to make it [Editorial]. :)

With that, I believe you'll be done!

**Comment by barnesstorming** [comment]
Sigh. You'd even told me to make sure to mark that "Editorial" ahead of time. So sorry. Anyway it's all fixed now, save the final modernize-spelling. When I run it, it only catches three things that I intentionally overrode in the past: 2 instances of "stone-walled" and one of "spread-eagled." In her uses, these are all compound adjectives. Mirriam-Webster has them both as one word only in verb form. So they should remain two words but hyphenated, right?

The context (emphasis added): 
Ch9: "...but a stalish pool in a **stone-walled depression** on the crest of a rise..."
Ch16: "For all the Terrans could learn, the population of the **stone-walled hive** might have been several ..."
Ch18: "...his wrists and ankles being made fast so that his body was **spread-eagled**."

**Comment by vr8hub** [comment]
Hmmm, good question. We'll leave it, and I'll let Alex decide what to do, and then we'll learn from him what to do going forward. :)

Great job, thanks for sticking with it!

**Comment by barnesstorming** [comment]
Literally could not have done it without you. Thanks for you patience and gentle nudging. I've learned much.
